### DISCUSSION TITLE
Automated/scripted view-splits, sorting specific buffer kinds into specific views...

### DISCUSSION THREAD

**Discussion by metaleap** [orbitalquark/textadept#149] [Q&A] [answered]
From my grokking of Textadept's (superbly rich) APIs, you clearly can't just say `thisBuffer.moveToView(thatView)`. Alright.

Still, I really gotta organize my workspace to keep editable&saveable "actual code files" distinctly separated and "together" from  all the debugger outputs, log messages, command-entry results, build outputs, test runs, find-references results etcpp. Ie. from all the "noise-of-doing"... just having a 2-views split would be sweet enough so that all files are in one and all "outputs" in the other. (But indeed automatic, not as incessant manual nudging and shoving.)

Now I could handle buffer&view creation&deletion events, and on event ensure there's exactly 2 views (ensuring it via repeated-unsplit and a single split), and then I'd need to sort the buffers to the view-they-belong:

But if that only works by closing-and-reopening them? --- once I (scriptually) delete a buffer
- if it's a file, I can re-open it in a specific view via goto_file, but at least its undo history is gone, possibly also last-caret-pos
- it it's a non-file programmatically-opened-and-populated buffer, I can hardly reopen it, at least it won't be reused by its originators the way it otherwise might

Or can I? Might I be missing some actually-latent or already-there capability in the APIs of achieving this?

**Comment by orbitalquark** [comment, marked as answer]
This area is definitely not one of Textadept's strengths. It's buffer/view organization capabilities are mostly non-existent. If you want to re-order buffers, you'd probably resort to modifying session files and then loading them (which is not recommended). And of course this would not work well for unsaved output buffers -- the buffers would be there, but initially empty.

**Reply by metaleap** [reply]
I noticed when I always just keep 2 split views open without premature impulsive unsplitting, that then all programmatic outputs seem to occur into the 2nd one regardless of focus, so that's quite alright actually. Nothing to do after all, sometimes it's so simple :grin: just when I had kept it stubbornly unsplit before at-all-times that it all felt unwieldy/intrusive.
